assets/js/modules/analytics/util/is-zero-report.test.js

```diff
@@ -0,0 +1,196 @@
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies.
+ */
+import { isZeroReport } from './is-zero-report';
+
+describe( 'isZeroReport', () => {
+	it( 'returns undefined when undefined is passed', () => {
+		expect( isZeroReport( undefined ) ).toBe( undefined );
+	} );
+
+	it.each( [
+		[ 'NULL', null ],
+		[ 'FALSE', false ],
+		[ 'a number', 1 ],
+		[ 'a string', 'test' ],
+		[ 'an empty object', {} ],
+		[ 'an object without rows or totals', [ { data: {} } ] ],
+		[ 'an object with empty rows', [ { data: { rows: [] } } ] ],
+		[ 'an object with empty totals', [ { data: { totals: [] } } ] ],
+	] )( 'returns true when %s is passed', ( _, report ) => {
+		expect( isZeroReport( report ) ).toBe( true );
+	} );
+
+	it( 'returns true id all totals are 0', () => {
+		const report = [
+			{
+				data: {
+					rows: [
+						{}, {}, {},
+					],
+					totals: [
+						{
+							values: [
+								'0',
+								'0',
+							],
+						},
+					],
+				},
+			},
+		];
+
+		expect( isZeroReport( report ) ).toBe( true );
+
+		const reportMultipleTotals = [
+			{
+				data: {
+					rows: [
+						{}, {}, {},
+					],
+					totals: [
+						{
+							values: [
+								'0',
+								'0',
+							],
+						},
+						{
+							values: [
+								'0',
+								'0',
+							],
+						},
+					],
+				},
+			},
+		];
+
+		expect( isZeroReport( reportMultipleTotals ) ).toBe( true );
+
+		const reportSingleTotal = [
+			{
+				data: {
+					rows: [
+						{}, {}, {},
+					],
+					totals: [
+						{
+							values: [
+								'0',
+							],
+						},
+					],
+				},
+			},
+		];
+
+		expect( isZeroReport( reportSingleTotal ) ).toBe( true );
+	} );
+
+	it( 'returns false for a report that has data', () => {
+		const report = [
+			{
+				data: {
+					rows: [
+						{}, {}, {},
+					],
+					totals: [
+						{
+							values: [
+								'123',
+								'234',
+							],
+						},
+					],
+				},
+			},
+		];
+
+		expect( isZeroReport( report ) ).toBe( false );
+
+		const reportMixedTotalValues = [
+			{
+				data: {
+					rows: [
+						{}, {}, {},
+					],
+					totals: [
+						{
+							values: [
+								'0',
+								'123',
+							],
+						},
+					],
+				},
+			},
+		];
+
+		expect( isZeroReport( reportMixedTotalValues ) ).toBe( false );
+
+		const reportMultipleTotals = [
+			{
+				data: {
+					rows: [
+						{}, {}, {},
+					],
+					totals: [
+						{
+							values: [
+								'123',
+								'234',
+							],
+						},
+						{
+							values: [
+								'0',
+								'345',
+							],
+						},
+					],
+				},
+			},
+		];
+
+		expect( isZeroReport( reportMultipleTotals ) ).toBe( false );
+
+		const reportSingleTotal = [
+			{
+				data: {
+					rows: [
+						{}, {}, {},
+					],
+					totals: [
+						{
+							values: [
+								'123',
+							],
+						},
+					],
+				},
+			},
+		];
+
+		expect( isZeroReport( reportSingleTotal ) ).toBe( false );
+	} );
```

Comment:
Please split the assertions in the last two tests here into separate tests. That is, those two currently have 3 expect calls which work fine as-is, but would be a bit more useful in the case of a failure to have separately with descriptive names for each.

Reply:
Both done and pushed.

Reply:
Thanks @benbowler – it looks like you only split the last test, can you do it for the previous one as well? Apologies if this wasn't clear from my last comment.